src/components/FilterChip/FilterChip.tsx

```diff
@@ -14,39 +14,43 @@ const FilterChipVariants = cva(
     },
   }
 );
-
-export interface FilterChipProps
-  extends React.HTMLAttributes<HTMLDivElement>,
-    VariantProps<typeof FilterChipVariants> {
+type addView = {
+  addView: true;
+  value: string;
+};
+type filterValue = {
+  addView: false;
   filter: string;
   value: string;
-  addView: boolean;
   onRemove?: () => void;
-}
+};
+export type FilterChipProps = React.HTMLAttributes<HTMLDivElement> &
+  VariantProps<typeof FilterChipVariants> &
+  (addView | filterValue);
 
 const FilterChip = ({
   className,
   addView,
-  filter,
   value,
-  onRemove,
   ...props
 }: FilterChipProps) => {
   return (
     <div className={cn(FilterChipVariants({ className, addView }))} {...props}>
```

Comment:
Dont' take verbatim, but something like this?

```suggestion
   let onRemoveProp;
  let restProps = props;
  
  if('onRemove' in props) {
    const { onRemove, ...otherProps } = props;
    onRemoveProp = onRemove;
    restProps = otherProps;
  }

  return (
    <div className={cn(FilterChipVariants({ className, addView }), className)} {...restProps}>
```